gtsfm/loader/one_d_sfm_loader.py

```diff
@@ -0,0 +1,162 @@
+"""Loader of dataset used in 1DSFM and Colmap paper
+
+The dataset is collected from internet and has EXIF embedded in each image,
+it contains several sequences with naming format: images.xxx/
+
+Folder sturcutre of a single sequence:
+- images/*.jpg
+
+Ref: https://www.cs.cornell.edu/projects/1dsfm/
+Paper 1: https://www.cs.cornell.edu/projects/1dsfm/docs/1DSfM_ECCV14.pdf
+Paper 2: https://openaccess.thecvf.com/content_cvpr_2016/papers/Schonberger_Structure-From-Motion_Revisited_CVPR_2016_paper.pdf
+
+Authors: Yanwei Du
+
+"""
+
+import glob
+import numpy as np
+import os
+from pathlib import Path
+from typing import List, Optional
+
+import scipy.io
+from gtsam import Cal3Bundler, Pose3
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.verification as verification_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class OneDSFMLoader(LoaderBase):
+    def __init__(
+        self,
+        folder: str,
+        image_extension: str = "jpg",
+        max_resolution: int = 640,
+        max_num_imgs: int = 0
+    ) -> None:
+        """Initializes to load from a specified folder on disk.
+
+        Args:
+            folder: the base folder which contains image sequence
+            image_extension: file extension for the image files. Defaults to 'jpg'.
+            max_resolution: integer representing maximum length of image's short side, i.e.
+                the smaller of the height/width of the image. e.g. for 1080p (1920 x 1080),
+                max_resolution would be 1080. If the image resolution max(height, width) is
+                greater than the max_resolution, it will be downsampled to match the max_resolution.
+            max_num_imgs: integer representing max number of images to process in the sequence,
+                0 or negative values means to process all the images
+        """
+        super().__init__(max_resolution=max_resolution)
+
+        # fetch all the file names in /images folder
+        search_path = os.path.join(folder, 'images', f"*.{image_extension}")
+
+        # @NOTE for now we only process images with valid EXIF data
+        self._image_paths, self._num_exif_imgs, self._num_all_imgs = \
+            self.get_images_with_valid_exif(search_path, max_num_imgs)
+        self._num_imgs = len(self._image_paths)
+
+        # self._image_paths.sort()
```

Comment:
nit: can we remove this commented out code?

Reply:
Done